### ISSUE TITLE
`DimensionMismatch` error

### ISSUE THREAD

**Issue by iharosi** [shellixyz/hd_fpv_video_tool#6]
Since Betaflight 4.4, the HD OSD layout is available and using these `.osd` files I'm getting the following error on cli:
<img width="1512" alt="Screenshot 2023-03-18 at 22 55 59" src="https://user-images.githubusercontent.com/473670/226142401-9ac51c6d-a45f-4efe-8634-1bf2ba67f47d.png">
It looks like it's some kind of a layout dimension mismatch issue.
I've tried the `generate-overlay-frames` and the `transcode-video` options but getting the same error.

Info about the `.osd` file:
```
INFO  > detected OSD file with DJI_SD tile layout

OSD file type: DJI FPV
Format version: 1
OSD size: 30x15 tiles
OSD tiles dimension: 36x54 px
OSD video offset: x: 180, y: 0 px
OSD Font variant: 1 (Betaflight)
Number of OSD frames: 3477
Highest video frame index: 22776
OSD update rate: 15% of the video frames (9.2Hz or approximately every 7 frames)
```

Strange thing it's been recognized as an SD file.
I'm not sure if the error is on wtfos, betaflight or hd_fpv_video_tool end. :S

**Comment by iharosi** [comment]
Info about the `.osd` file:
```
INFO  > detected OSD file with DJI_SD tile layout

OSD file type: DJI FPV
Format version: 1
OSD size: 30x15 tiles
OSD tiles dimension: 36x54 px
OSD video offset: x: 180, y: 0 px
OSD Font variant: 1 (Betaflight)
Number of OSD frames: 3477
Highest video frame index: 22776
OSD update rate: 15% of the video frames (9.2Hz or approximately every 7 frames)
```

**Comment by shellixyz** [comment]
It must be the size of OSD which is not handled. Could you send the OSD file (and video file optionally) to me so that I can test with the actual data ? You can send it in private if you don't want to make the data public: `@shellixyz` on Telegram or via email `shellixyz@gmail.com`

**Comment by iharosi** [comment]
Hi,
I've sent the files on email.
Thanks for your help!

**Comment by shellixyz** [comment]
I investigated the issue. It turns out it is a bug in WTF.FPV's msp-osd which doesn't write the right OSD size in the header for some reason. I modified the code to display a warning instead of crashing if it is the case. Still won't render the OSD right but at least won't crash. I could add an option to force the size of the OSD but the right solution would be to fix msp-osd's code if it hasn't been already.
